airflow/serialization/serialized_objects.py

```diff
@@ -253,7 +253,8 @@ def _value_is_hardcoded_default(cls, attrname: str, value: Any) -> bool:
         default. (This is because ``"default" is "default"`` will be False as
         they are different strings with the same characters.)
         """
-        if attrname in cls._CONSTRUCTOR_PARAMS and cls._CONSTRUCTOR_PARAMS[attrname].default is value:
+        if attrname in cls._CONSTRUCTOR_PARAMS and \
+                (cls._CONSTRUCTOR_PARAMS[attrname].default is value or (value in [{}, []])):
```

Comment:
If something has a default of None and a user explicitly passes in None, does this change now mean we store it sa `null` in the JSON?

Reply:
No it is excluded because of the following function:

https://github.com/apache/airflow/blob/e6f2e19912a1c588c79bb8ac137d09d6789d8126/airflow/serialization/serialized_objects.py#L107-L114